classes/module/Module.php

```diff
@@ -1380,7 +1380,7 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
                     $file = trim(file_get_contents(_PS_MODULE_DIR_ . $module . '/' . $module . '.php'));
 
                     try {
-                        $parser = (new PhpParser\ParserFactory())->create(PhpParser\ParserFactory::PREFER_PHP7);
+                        $parser = (new PhpParser\ParserFactory())->create(PhpParser\ParserFactory::ONLY_PHP7);
```

Comment:
We don't support anymore PHP5.6, I don't think it's a BC breaks :thinking:

Reply:
> Does it mean a module not compatible with PHP7 will now be considered invalid?

I think it's consistent. "My software works with vendor X > v7, then I only accept plugins compatible with X > v7". We have regularly complains about bad modules in the wild, I think we should aim to discard such modules as soon as possible (before install happens).

> at the very least it's probably worth mentioning as an important change for this version and needs to be documented no?

I 👍  . It is better to clear the confusion.

Reply:
This should be noted

Reply:
It will be in the release note and the devdoc of course

Reply:
I totally agree with the updated constraint, it totally makes sense But then it is a BC break for modules since a code not PHP7 compliant which used to work will be rejected This alone should justify to note this PR as a BC break
Although it's perfectly understandable and acceptable

Reply:
We already did it when we removed the PHP5.6 support. So if a module is not compliant with PHP7, it shouldn't be able to run since the 1.7.6.0 :thinking: